### PR TITLE
Add documentation for REST API

### PR DESCRIPTION
**Pull request by mavam**
This PR adds documentation for the REST API.

Note that I had to downgrade the Python package `ipykernel` because the latest version was yanked from PyPI, causing the docs build to fail.

### CODE REVIEW

File: web/docs/use/integrate/rest-api/README.md [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+```
+
+For local testing and development, generating suitable certificates and tokens
+can be a hassle. For this scenario, you can start the server in [developer
+mode](#developer-mode) where it accepts plain HTTP connections are does not
+perform token authentication.
+
+## TLS Modes
+
+There exist 4 modes to start the REST API, each of which suits a slightly
```
Comment (dominiklohmann) [inline web/docs/use/integrate/rest-api/README.md:66]:
```suggestion
There exist four modes to start the REST API, each of which suits a slightly
```

**Review by dominiklohmann — APPROVED** [review]
Fancy graphics! ✨

**Review by lava — APPROVED** [review]
Overall everything looks good to me.

I'm a bit skeptical about the "scaling" part; running REST API instances running in parallel (can't comment on them directly), which is more of a side-effect of how the actor system works rather than something we designed for explicitly or ever tested, and I think it's unlikely that this would actually help in most scenarios since the server itself does very little work apart from translating the http requests into actor system calls.

File: web/docs/use/integrate/rest-api/README.md [hunk 2]
```diff
@@ -0,0 +1,146 @@
+---
+sidebar_position: 0
+sidebar_label: REST API
+---
+
+# REST API
+
+VAST provides a REST API to interact with a server node. The `web`
+[plugin](/docs/understand/architecture/plugins) implements the API as a
+dedicated component that can run in multiple ways.
```
[the remark this answers is not in the document]
Comment (mavam) [reply web/docs/use/integrate/rest-api/README.md:10]: I added a sentence about that.

**Comment by mavam** [comment]
@lava I added a note that the scaling is something we get "for free" because the system architecture, but in practice will likely be irrelevant.

**Pull request merged**